hnn_core/opt_toy_example/optimize_evoked.py

```diff
@@ -0,0 +1,164 @@
+                         sigma=params_dict['evprox1_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p1,
+                         synaptic_delays=synaptic_delays_p)
+
+    # Distal
+    weights_ampa_d1 = {'L2_basket': params_dict['evdist1_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evdist1_ampa_L2_pyramidal'],
+                       'L5_pyramidal': params_dict['evdist1_ampa_L5_pyramidal']}
+    weights_nmda_d1 = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
+                       'L5_pyramidal': 0.080074}
+    synaptic_delays_d1 = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                          'L5_pyramidal': 0.1}
+
+    net.add_evoked_drive('evdist1',
+                         mu=params_dict['evdist1_mu'],
+                         sigma=params_dict['evdist1_sigma'],
+                         numspikes=1,
+                         location='distal',
+                         weights_ampa=weights_ampa_d1,
+                         weights_nmda=weights_nmda_d1,
+                         synaptic_delays=synaptic_delays_d1)
+
+    # Proximal 2
+    weights_ampa_p2 = {'L2_basket': params_dict['evprox2_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evprox2_ampa_L2_pyramidal'],
+                       'L5_basket': params_dict['evprox2_ampa_L5_basket'],
+                       'L5_pyramidal': params_dict['evprox2_ampa_L5_pyramidal']}
+
+    net.add_evoked_drive('evprox2',
+                         mu=params_dict['evprox2_mu'],
+                         sigma=params_dict['evprox2_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p2,
+                         synaptic_delays=synaptic_delays_p)
+
+###############################################################################
+# Then, we define the constrainst.
+#
+# The constraints must be a dictionary of tuples where the first value in each
+# tuple is the lower bound and the second value is the upper bound for the
+# corresponding parameter.
+
+
+constraints = dict()
+constraints.update({'evprox1_ampa_L2_basket': (0.01, 1.),
+                    'evprox1_ampa_L2_pyramidal': (0.01, 1.),
+                    'evprox1_ampa_L5_basket': (0.01, 1.),
+                    'evprox1_ampa_L5_pyramidal': (0.01, 1.),
+                    'evprox1_mu': (5., 50.),
+                    'evprox1_sigma': (2., 25.),
+                    'evdist1_ampa_L2_basket': (0.01, 1.),
+                    'evdist1_ampa_L2_pyramidal': (0.01, 1.),
+                    'evdist1_ampa_L5_pyramidal': (0.01, 1.),
+                    'evdist1_mu': (50., 80.),
+                    'evdist1_sigma': (2., 25.),
+                    'evprox2_ampa_L2_basket': (0.01, 1.),
+                    'evprox2_ampa_L2_pyramidal': (0.01, 1.),
+                    'evprox2_ampa_L5_basket': (0.01, 1.),
```

Comment:
some rationale should be given why these bounds were chosen

Reply:
Perhaps let's save refining the example for a future PR? Especially since the API is subject to change